junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -137,12 +139,19 @@ public ScriptPair call() throws Exception {
                     case FINISHED:
                         FinishedEvent finished = (FinishedEvent) event;
                         // note: observed script is possibly incomplete
-                        String observedScript = finished.getScript();
-                        return new ScriptPair(expectedScript, observedScript);
+                        observedScript = finished.getScript();
+                        
+                        // we keep the result and read any other event is still in the buffer. We will return after the first timeout
+//                        return new ScriptPair(expectedScript, observedScript);
+                        finishedReceived = true;
+                        break;
                     default:
                         throw new IllegalArgumentException("Unrecognized event kind: " + event.getKind());
                     }
                 } catch (SocketTimeoutException e) {
+                    if (finishedReceived)
```

Comment:
> The problem is that we can still receive events from the driver after the FINISHED events,  

Is that a spec violation by the driver?  Why not fix that?  Otherwise, we need this workaround on every client.

Waiting on the SocketTimeoutException seems to be a hack. Would greatly prefer another approach.  Waiting for confirmation on the `Disposed` is necessary as otherwise there is a race between tests on clean up of the driver and potential attempts to reuse that resource on the next test